nistats/second_level_model.py

```diff
@@ -442,3 +443,134 @@ def compute_contrast(
+            for model_idx, niimg in enumerate(second_level_input):
+                if not isinstance(niimg, (str, Nifti1Image)):
+                    raise ValueError(' object at idx %d is %s instead of'
+                                     ' Niimg-like object' %
+                                     (model_idx, type(niimg)))
+    elif isinstance(second_level_input, (str, Nifti1Image)):
+        if design_matrix is None:
+            raise ValueError('List of niimgs as second_level_input'
+                             ' require a design matrix to be provided')
+        second_level_input = check_niimg(niimg=second_level_input,
+                                         ensure_ndim=4)
+    else:
+        raise ValueError('second_level_input must be'
+                         ' a list Niimg-like objects. Instead %s '
+                         'was provided' % type(second_level_input))
+
+    # Check confounds
+    if confounds is not None:
+        if not isinstance(confounds, pd.DataFrame):
+            raise ValueError('confounds must be a pandas DataFrame')
+        if 'subject_label' not in confounds.columns:
+            raise ValueError('confounds DataFrame must contain column'
+                             '"subject_label"')
+        if len(confounds.columns) < 2:
+            raise ValueError('confounds should contain at least 2 columns'
+                             'one called "subject_label" and the other'
+                             'with a given confound')
+        # Make sure subject_label contain strings
+        labels_index = confounds.columns.tolist().index('subject_label')
+        labels_dtype = confounds.dtypes[labels_index]
+        if not isinstance(labels_dtype, np.object):
+            raise ValueError('subject_label column must be of dtype '
+                             'object instead of dtype %s' % labels_dtype)
+
+    # Check design matrix
+    if not isinstance(design_matrix, pd.DataFrame):
+            raise ValueError('design matrix must be a pandas DataFrame')
+
+    # Report progress
+    t0 = time.time()
+    if verbose > 0:
+        sys.stderr.write("Fitting second level model. "
+                         "Take a deep breath\r")
+
+    # Select sample map for masker fit and get subjects_label for design
+    sample_map = mean_img(second_level_input)
+
+    # Learn the mask. Assume the first level imgs have been masked.
+    if not isinstance(mask, NiftiMasker):
+        masker = NiftiMasker(
+            mask_img=mask, smoothing_fwhm=smoothing_fwhm,
+            memory=memory, verbose=max(0, verbose - 1),
+            memory_level=memory_level)
+    else:
+        masker = clone(mask)
+        # TODO: manage changing mask parameters
+    masker.fit(sample_map)
+
+    # Report progress
+    if verbose > 0:
+        sys.stderr.write("\nComputation of second level model done in "
+                         "%i seconds\n" % (time.time() - t0))
+
+    # Check or obtain the contrast
```

Comment:
Could go in a side contrast (and should also be  used in SecondLevelModel)

Reply:
Done